src/meta/src/manager/catalog/fragment.rs

```diff
@@ -1411,18 +1408,12 @@ impl FragmentManager {
             let table_fragments = map
                 .get(&table_id)
                 .with_context(|| format!("table_fragment not exist: id={}", table_id))?;
-            match table_job_type.as_ref() {
-                Some(TableJobType::SharedCdcSource) => {
-                    if let Some(fragment) = table_fragments.source_fragment() {
-                        fragments.insert(table_id, fragment);
-                    }
-                }
-                // MV on MV, and other kinds of table job
-                None | Some(TableJobType::General) | Some(TableJobType::Unspecified) => {
-                    if let Some(fragment) = table_fragments.mview_fragment() {
-                        fragments.insert(table_id, fragment);
-                    }
-                }
+
+            if let Some(fragment) = table_fragments.mview_fragment() {
+                fragments.insert(table_id, fragment);
+            } else if let Some(fragment) = table_fragments.source_fragment() {
+                // look for Source fragment if there's no MView fragment
+                fragments.insert(table_id, fragment);
```

Comment:
This is quite counter-intu😄tive. May add some comments to show that this is for `Source` job?

Reply:
Oh, it's already on `pub async fn get_upstream_root_fragments`.